### PR TITLE
[BE001] Users Migration

### PR DESCRIPTION
**Pull request by aetheryna**
## What happened

Created the migration file for the user's table. Included the entity file for NestJS referencing the DB. 👷

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+## Database
+PG_DB=crit_dev
+PG_PORT=5432
+PG_USERNAME=postgres
+PG_PASSWORD=1234
+PG_HOST=localhost
```
Comment (bterone) [inline .env]: Let's specify all these in the `.env.development` file instead so we know its for development only 👍 

We can set a custom path to the ENV file to the development env for now by adding one rule https://docs.nestjs.com/techniques/configuration#custom-env-file-path

File: src/app.module.ts [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
 
+import { SeedsModule } from './modules/seeds/seeds.module'
+
+import { SeedsService } from '@services/seeds.service';
+
 @Module({
   imports: [
     ConfigModule.forRoot({
-      envFilePath: [`.env.${process.env.NODE_ENV}`, '.env']
+      envFilePath: ['../env']
```
Comment (bterone) [inline src/app.module.ts]: In this case, we could just do the following right? That way we can keep the `.env` variables inside the `.env.development` file? 🤔 

```suggestion
      envFilePath: [`../.env.${process.env.NODE_ENV}`, '../.env']
```

File: src/db/migrations/1649748613752-CreateUsers.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          name: "user_id",
+          type: "uuid",
+          isGenerated: true,
+          default: "gen_random_uuid()",
+          isPrimary: true
+        },
+        {
+          name: "firstName",
+          type: "varchar",
+          isNullable: true
```
Comment (bterone) [inline src/db/migrations/1649748613752-CreateUsers.ts]: First/Last name doesn't need to be nullable 😂

File: src/services/seeds.service.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@nestjs/common'
+import { EntityManager } from 'typeorm/entity-manager/EntityManager'
+import { User } from '../entities/user.entity'
+
+@Injectable()
+export class SeedsService {
+  constructor (private readonly entityManager: EntityManager) {}
+
+  async perform() :Promise<void> {
```
Comment (bterone) [inline src/services/seeds.service.ts]:
```suggestion
  async perform(): Promise<void> {
```

File: src/services/users.service.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@nestjs/common'
+import { InjectRepository } from '@nestjs/typeorm'
+import { Repository } from 'typeorm'
+import { User } from 'src/entities/user.entity'
+
+@Injectable()
+export class UsersService {
```
Comment (bterone) [inline src/services/users.service.ts]: Let's write tests for each of the methods in this file 👍

**Review by bterone — COMMENTED** [review]
Make sure to run `npm run format` btw, there's quite a few linting errors 👍

File: src/services/users.service.ts (hunk 5 above)
Comment (aetheryna) [reply src/services/users.service.ts]: That's all the sample code, deleted the file 💸

**Review by bterone — APPROVED** [review]

**Pull request merged**